Uplift/Controllers/SportsDetailViewController.swift

```diff
@@ -9,23 +9,181 @@
+        let headerView = collectionView.dequeueReusableSupplementaryView(
+            ofKind: UICollectionView.elementKindSectionHeader,
+            withReuseIdentifier: sportsDetailHeaderViewReuseIdentifier,
+            for: indexPath) as! SportsDetailHeaderView
+        headerView.configure(for: self, for: post)
+        return headerView
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let itemType = section.items[indexPath.item]
+        switch itemType {
+        case .info:
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: infoSectionReuseIdentifier, for: indexPath) as! SportsDetailInfoCollectionViewCell
+            cell.configure(for: post)
+            return cell
+        case .players:
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: playersSectionReuseIdentifier, for: indexPath) as! SportsDetailPlayersCollectionViewCell
+            cell.configure(for: post, dropStatus: dropStatus)
+            return cell
+        case .discussion:
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: discussionSectionReuseIdentifier, for: indexPath) as! SportsDetailDiscussionCollectionViewCell
+            cell.configure(for: post)
+            return cell
+        case .comment(let c):
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: commentSectionReuseIdentifier, for: indexPath) as! SportsDetailCommentCollectionViewCell
+            cell.configure(for: c)
+            return cell
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, referenceSizeForHeaderInSection section: Int) -> CGSize {
+        return CGSize(width: collectionView.frame.width, height: 264)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        return 0
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let itemType = section.items[indexPath.item]
+        let width = collectionView.frame.width
+
+        switch itemType {
+        case .info:
+            return CGSize(width: width, height: 157)
+        case .players:
+            // TODO: add player names field to post.
+            let players = [User(id: "p1", name: "Zain Khoja", netId: "znksomething"),
+            User(id: "p2", name: "Amanda He", netId: "noidea"),
+            User(id: "p3", name: "Yi Hsin Wei", netId: "y???")]
+            let nameHeight = 24
+            let baseHeight = 82
+            let height = CGFloat(baseHeight + nameHeight * (dropStatus == .closed ? 1 : players.count))
```

Comment:
can we put `dropStatus == .closed ? 1 : players.count` into a constant. this line is kind of hard to read atm